### PR TITLE
improve xss security

### LINKED ISSUE #838: Security Cleanup
@yotamberk For details about this issue see email.

### PR DESCRIPTION
**Pull request by mojoaxel**
I added the [xss library](https://www.npmjs.com/package/xss) to improve security.

fixes #838

### CODE REVIEW

File: lib/shared/Popup.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ class Popup {
       this.frame.appendChild(content);
     }
     else {
-      this.frame.innerHTML = content; // string containing text or HTML
+      this.frame.innerHTML = util.xss(content); // string containing text or HTML
```
Comment (yotamberk) [inline lib/shared/Popup.js:45]: util is not imported

File: lib/timeline/component/CustomTime.js [hunk 2]
```diff
@@ -205,7 +205,7 @@ class CustomTime extends Component {
   setCustomMarker(title, editable) {
     const marker = document.createElement('div');
     marker.className = `vis-custom-time-marker`;
-    marker.innerHTML = title;
+    marker.innerHTML = util.xss(title);
```
Comment (yotamberk) [inline lib/timeline/component/CustomTime.js:208]: here too

**Review by yotamberk — CHANGES_REQUESTED** [review]
In some of the files `util` is not imported

File: lib/shared/Popup.js (hunk 1 above)
Comment (mojoaxel) [reply lib/shared/Popup.js:46]: fixed

**Review by yotamberk — APPROVED** [review]

**Comment by Donov4n** [comment]
Hello :)

Its seems that this PR has introduced a regression because the `xss` dep. has only been added   
to devDeps and is therefore not available when consuming this vis-timeline.

The error : 

```
This dependency was not found:

* xss in ./node_modules/vis-timeline/esnext/esm/vis-timeline-graph2d.js

To install it, you can run: npm install --save xss
```

Should i open a new issue for this ?

**Comment by Thomaash** [comment]
Hi there,

I know this is already out as a fix but this is a breaking change. XSS is basically nuking the HTML away with an extremely constrained set of exceptions. There are no forms, no classes, no styles etc. Also the thing has like 30K and for what? So that people can have the convenience of `<h2>Heading</h2><p>Text…</p>`? This doesn't seem worth it to me (though as always, this is up for discussion). If we really want to sanitize this and since we allow for HTMLElement to be used as a title, I would just use innerText for strings and changed the docs so that the people who want HTML in there would be instructed to pass HTMLElement (maybe using innerHTML there, if they choose to) instead of a string.

Another thing is that the configurator (which is except for reformatting identical to the code in Vis Network and really should be moved to Vis Util) doesn't really use HTML (apart from some simple wrapping, but we don't need innerHTML for that), it can be quite easily converted to innerText and no sanitation is necessary at all.

**Comment by mojoaxel** [comment]
@Donov4n Sorry! 
@Thomaash Thanks!

**Comment by ckeboss** [comment]
I agree with @Thomaash, shouldn't sanitization be up to the user?

For example, setting a tooltip template is now pretty useless, as you can not input any attributes to elements.

**Pull request merged**